src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -201,16 +191,12 @@ public class XmlConvert
             int position = 0;
 
             int underscorePos = name.IndexOf('_');
-            MatchCollection? mc = null;
             IEnumerator? en = null;
             if (underscorePos >= 0)
             {
-                if (s_encodeCharPattern == null)
-                {
-                    s_encodeCharPattern = new Regex("(?<=_)[Xx]([0-9a-fA-F]{4}|[0-9a-fA-F]{8})_");
-                }
+                s_encodeCharPattern ??= new Regex("(?<=_)[Xx]([0-9a-fA-F]{4}|[0-9a-fA-F]{8})_");
 
-                mc = s_encodeCharPattern.Matches(name, underscorePos);
+                MatchCollection? mc = s_encodeCharPattern.Matches(name, underscorePos);
```

Comment:
is this actually nullable? I'd expect you get warning on the next line, given that you haven't I assume it's non-null